packages/flame_test/lib/src/fails_assert.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:test/test.dart';
+
+/// Matcher that can be used in a test that expects an assertion error.
+///
+/// This is similar to standard `throwsAssertionError` matcher, but also
+/// allows an optional message string to verify that the assertion has the
+/// expected message.
+///
+/// For example:
+/// ```dart
+/// expect(
+///   () => PositionComponent(size: Vector2.all(-1)),
+///   failsAssert('size of a PositionComponent cannot be negative'),
+/// )
+/// ```
+Matcher failsAssert([String? message]) {
```

Comment:
```suggestion
Matcher failsWithMessage([String? message]) {
```

Reply:
I wanted to have some kind of indication that this is an `AssertionError`, not any other kind of error. And eventually we would want to have similar matchers for other types of errors as well.

Reply:
I love this!!!